sdk-extensions/metric-incubator/src/main/java/io/opentelemetry/sdk/viewconfig/ViewConfig.java

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.viewconfig;
+
+import static java.util.stream.Collectors.toList;
+
+import io.opentelemetry.sdk.autoconfigure.spi.ConfigurationException;
+import io.opentelemetry.sdk.metrics.SdkMeterProviderBuilder;
+import io.opentelemetry.sdk.metrics.common.InstrumentType;
+import io.opentelemetry.sdk.metrics.view.Aggregation;
+import io.opentelemetry.sdk.metrics.view.InstrumentSelector;
+import io.opentelemetry.sdk.metrics.view.MeterSelector;
+import io.opentelemetry.sdk.metrics.view.View;
+import io.opentelemetry.sdk.metrics.view.ViewBuilder;
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import javax.annotation.Nullable;
+import org.yaml.snakeyaml.Yaml;
+
+/**
+ * Enables file based YAML configuration of Metric SDK {@link View}s.
+ *
+ * <p>For example, a YAML file with the following content:
+ *
+ * <pre>
+ *   - selector:
+ *       instrument_name: my-instrument
+ *       instrument_type: COUNTER
+ *       meter_name: my-meter
+ *       meter_version: 1.0.0
+ *       meter_schema_url: http://example.com
+ *     view:
+ *       name: new-instrument-name
+ *       description: new-description
+ *       aggregation: histogram
+ *       attribute_keys:
+ *         - foo
+ *         - bar
+ * </pre>
+ *
+ * <p>Is equivalent to the following configuration:
+ *
+ * <pre>{@code
+ * SdkMeterProvider.builder()
+ *     .registerView(
+ *         InstrumentSelector.builder()
+ *             .setInstrumentName("my-instrument")
+ *             .setInstrumentType(InstrumentType.COUNTER)
+ *             .setMeterSelector(
+ *                 MeterSelector.builder()
+ *                     .setName("my-meter")
+ *                     .setVersion("1.0.0")
+ *                     .setSchemaUrl("http://example.com")
+ *                     .build())
+ *             .build(),
+ *         View.builder()
+ *             .setName("new-instrument")
+ *             .setDescription("new-description")
+ *             .setAggregation(Aggregation.histogram())
+ *             .filterAttributes(key -> new HashSet<>(Arrays.asList("foo", "bar")).contains(key))
+ *             .build());
+ * }</pre>
+ */
+public class ViewConfig {
+
+  private ViewConfig() {}
+
+  /**
+   * Load the view configuration YAML from the file and apply it to the {@link
+   * SdkMeterProviderBuilder}.
+   *
+   * @throws ConfigurationException if unable to interpret file contents
+   */
+  @SuppressWarnings("NullAway")
```

Comment:
Perhaps something changed in an iteration of the PR but the methods being `assert`ed against null don't seem to be marked `Nullable` so I guess we don't need this or those assertions

Reply:
For reference, nullaway will recognize `requireNonNull` but not `assert` since it is trying to ensure runtime safety (I wish it did check asserts personally though). So using it would be better than suppressing where it would be needed

Reply:
Yes, with the change to snakeyaml we're get stronger non-null guarantees than with jackson (at the expense of manual parsing code).